### PR TITLE
Post Editor: Wire up the dialog opt-in button to set the blog's editor to Gutenberg. [WIP]

### PR DESCRIPTION
**Pull request by kwight**
Upcoming API work will add an endpoint for setting a blog to use the Gutenberg editor. This PR stubs out an action to consider while the API work happens.

**Testing instructions**

- Load `/post/:site` and then "Learn more" in the sidebar's Gutenberg visual.
- In the dialog, click the "Try the new editor" button.
- Verify a successful call to `TBD`.
- Verify the test site has the correct blog sticker.

### CODE REVIEW

**Comment by kwight** [comment]
@dmsnell I've updated this to use an action more in-line with our API discussions today – let me know if I've misunderstood, and feel free to push if you know more details! 👍

File: client/state/action-types.js [hunk 1]
```diff
@@ -212,6 +212,7 @@ export const EDITOR_SAVE = 'EDITOR_SAVE';
 export const EDITOR_EDIT_RAW_CONTENT = 'EDITOR_EDIT_RAW_CONTENT';
 export const EDITOR_INIT_RAW_CONTENT = 'EDITOR_INIT_RAW_CONTENT';
 export const EDITOR_RESET_RAW_CONTENT = 'EDITOR_RESET_RAW_CONTENT';
+export const EDITOR_SET_GUTENBERG = 'EDITOR_SET_GUTENBERG';
```
Comment (dmsnell) [inline client/state/action-types.js]: we might want to double-check our naming here. if we are going to offer opt-out then something like `EDITOR_SELECT` or `EDITOR_OPT_IN` which takes a parameter specifying the editor experience might be more clear in its intent

**Comment by dmsnell** [comment]
@kwight this is going in the direction yeah that I'd expect. let's connect on Monday and check again after putting in the API

having a Redux action is a good idea and probably necessary so we can adjust other UI concerns in Calypso about it.

**Comment by kwight** [comment]
Replaced by #27893 .

**Pull request closed without merging**